---
identifier: "CVE-2021-29569"
package_slug: "maven/org.tensorflow/parentpom"
title: "Out-of-bounds Read"
description: "TensorFlow is an end-to-end open source platform for machine learning.
  The implementation of `tf.raw_ops.MaxPoolGradWithArgmax` can cause reads outside
  of bounds of heap allocated data if attacker supplies specially crafted inputs.
  The implementation(https://github.com/tensorflow/tensorflow/blob/ac328eaa3870491ababc147822cd04e91a790643/tensorflow/core/kernels/requantization_range_op.cc#L49-L50)
  assumes that the `input_min` and `input_max` tensors have at least one element,
  as it accesses the first element in two arrays. If the tensors are empty, `.flat<T>()`
  is an empty object, backed by an empty array. Hence, accesing even the 0th element
  is a read outside the bounds. The fix will be included in TensorFlow We will also
  cherrypick this commit on TensorFlow, TensorFlow, TensorFlow and TensorFlow, as
  these are also affected and still in supported range."
date: "2021-05-19"
pubdate: "2021-05-14"
affected_range: "(,2.1.4),[2.2.0,2.2.3),[2.3.0,2.3.3),[2.4.0,2.4.2)"
fixed_versions: []
affected_versions: "All versions before 2.1.4, all versions starting from 2.2.0 before
  2.2.3, all versions starting from 2.3.0 before 2.3.3, all versions starting from
  2.4.0 before 2.4.2"
not_impacted: ""
solution: "Unfortunately, there is no solution available yet."
urls:
- "https://nvd.nist.gov/vuln/detail/CVE-2021-29569"
cvss_v2: "AV:L/AC:L/Au:N/C:P/I:N/A:P"
cvss_v3: "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:L/I:N/A:L"
uuid: "e06d46f3-59f3-4e65-bd28-c2d3f858e252"
